choreographer/src/main/java/eu/arrowhead/core/choreographer/database/service/ChoreographerDBService.java

```diff
@@ -79,33 +98,45 @@
     @Autowired
     private ChoreographerRunningStepRepository choreographerRunningStepRepository;
 
+    @Autowired
+    private ChoreographerExecutorRepository choreographerExecutorRepository;
+
+    @Autowired
+    private ChoreographerStepDetailRepository choreographerStepDetailRepository;
+
+    @Autowired
+    private ChoreographerExecutorServiceDefinitionRepository choreographerExecutorServiceDefinitionRepository;
+
+    @Autowired
+    private ChoreographerExecutorServiceDefinitionConnectionRepository choreographerExecutorServiceDefinitionConnectionRepository;
+
     private final Logger logger = LogManager.getLogger(ChoreographerDBService.class);
     
     //=================================================================================================
 	// methods
 
     //-------------------------------------------------------------------------------------------------
     @Transactional(rollbackFor = ArrowheadException.class)
-    public ChoreographerStep createStep(final String name, final String serviceName, final String metadata, final String parameters, final int quantity, final long actionId) {
+    public ChoreographerStep createStep(final String name, final String metadata, final String parameters, final int quantity, final long actionId) {
         logger.debug("createStep started...");
 
         try {
             if (Utilities.isEmpty(name)) {
                 throw new InvalidParameterException("Step name is null or blank.");
             }
 
-            if (Utilities.isEmpty(serviceName)) {
-                throw new InvalidParameterException("Service name is null or blank.");
-            }
-
             final Optional<ChoreographerStep> stepOptional = choreographerStepRepository.findByNameAndActionId(name, actionId);
             stepOptional.ifPresent(step -> {
                 throw new InvalidParameterException("Action already has a step with the same name. Step names must be unique in each Action.");
             });
```

Comment:
this check is unnecessary, because in createPlan, we already check for duplicates.